terror/terror.go

```diff
@@ -257,7 +257,7 @@ func (e *Error) NotEqual(err error) bool {
 // ToSQLError convert Error to mysql.SQLError.
 func (e *Error) ToSQLError() *mysql.SQLError {
 	code := e.getMySQLErrorCode()
-	return mysql.NewErrf(code, e.getMsg())
+	return mysql.NewErrf(code, "%s", e.getMsg())
```

Comment:
@shenli `mysql.NewErr` cannot be used here. It's functionality is not what we usually expected to be:

mysql.NewErr: NewErr generates a SQL error, with an error code and default format specifier defined in MySQLErrName.

```go
func NewErr(errCode uint16, args ...interface{}) *SQLError
```

mysql.NewErrf: NewErrf creates a SQL error, with an error code and a format specifier.

```go
func NewErrf(errCode uint16, format string, args ...interface{}) *SQLError
```